### DISCUSSION TITLE
Can someone please help with a simple plugin development ?

### DISCUSSION THREAD

**Discussion by Its-my-right** [koreader/koreader#12059] [Q&A]
Hi everyone,

I wrote a simple lua plugin for koreader, enabling a "double screen refresh" when reading.

This can help with reading color mangas/comics to limit ghosting.

But, I would like the plugin to do something only if what is actually displayed is a color image. (At the moment, the "double refresh" is always active, even on text or black and white images).

And I don't know how to do that. My color files in cbz archives contains "_color" in their names so I thought instead of analysing the picture pixel by pixel (probably a bad idea on a e-reader), simply checking the name of the image currently displayed would be a "good" way to go, but I have no idea how to get the name of the image displayed in my lua plugin.

If someone could help me on this one, it would be very nice 

Thanks

**Comment by pazos** [comment]
Xref: https://www.mobileread.com/forums/showthread.php?t=361963

**Comment by AryaTwirl** [comment]
Hi, May I try out your extension? It is due to the fact that my Kobo Libra 2 still exhibits slight ghosting even after refresh every page, so I want to see whether your plugin can help ease my issue.

Thank you.

**Comment by NiLuJe** [comment]
Generally, if the top-level widget has a field named `dithered` set to `true`, there's a fair chance there's image content in there.

That's how we handle, you guessed it, hardware dithering, and what we piggy-back on for Kaleido waveform modes, too.

In the case of the reader, that would be ReaderUI, c.f., https://github.com/koreader/koreader/blob/7e52c15de29c013413f3d9ffef37d2de86b44dbf/frontend/apps/reader/readerui.lua#L134-L141 & https://github.com/koreader/koreader/blob/7e52c15de29c013413f3d9ffef37d2de86b44dbf/frontend/apps/reader/modules/readerview.lua#L261-L287
